### ISSUE TITLE
ValueError: 2 is not a valid Method

### ISSUE THREAD

**Issue by IAmAbszol** [hohav/py-slippi#8]
**Python Version**: 3.6.8
**Slippi Version**: 1.2.0

**How to reproduce**
1. Install the dependencies + termcolor as it's currently not contained within the requirements.txt.
2. Grab latest Slippi files, mine was tested on any HungryBox vs. Mang0 set at Pound 2019 (Specifically Grandfinals).
3. Import Slippi module or just bring in the Game class.
4. Error will be thrown whenever you invoke the constructors path to the Slippi stream file.

**Error**
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\site-packages\slippi\game.py", line 33, in __init__
    self._parse_file(path)
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\site-packages\slippi\game.py", line 85, in _parse_file
    event = self._parse_event(stream, payload_sizes)
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\site-packages\slippi\game.py", line 68, in _parse_event
    return evt.End._parse(stream)
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\site-packages\slippi\event.py", line 192, in _parse
    return cls(cls.Method(method))
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\enum.py", line 293, in __call__
    return cls.__new__(cls, value)
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\enum.py", line 535, in __new__
    return cls._missing_(value)
  File "C:\Users\Kyle\Miniconda3\envs\meleeai\lib\enum.py", line 548, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 2 is not a valid Method
```

The error resides within the **event.py** line **192**.
The **method** variable is of type int with value 2 which doesn't match to Methods 0 or 3 being Inconclusive or Conclusive (Respectfully).

**Temporary Fix**
I haven't played with Python's struct pack/unpack functions too much but from what I gather, the 2 being returned has something to do with how the file was packed in the first place. Thus I further assume that the creator of the Slippi file has changed it's packing parameters?

**Source(s)**: [API](https://docs.python.org/2/library/struct.html#struct.pack) and [example](https://www.geeksforgeeks.org/struct-module-python/)

Therefore, changing the method variable from 2 to 3 since possibly 0 could still be used to solve the issue though I felt grimy for doing so.

Final code result within **event.py**
**Before**
```
    @classmethod
    def _parse(cls, stream):
        (method,) = unpack('B', stream)
        return cls(cls.Method(method))
``` 

**After**
```
    @classmethod
    def _parse(cls, stream):
        (method,) = unpack('B', stream)
        method = 3 if method == 2 else 0
        return cls(cls.Method(method))
```

**Comment by vladfi1** [comment]
I am seeing this as well on the Pound data.

**Comment by IAmAbszol** [comment]
@vladfi1 Glad Im not the only one but trying to find documentation on file changes is like findin a needle in a haystack

**Comment by hohav** [comment]
Thanks for the report. The issue is caused by [a change in Slippi 2.0.0](https://github.com/project-slippi/project-slippi/wiki/Replay-File-Spec/_compare/d3ba3fb88f81e2bb972ff627833473a2fba47f80...1c19dd368d1dd13c8dbabd8e714c6454c138f241#diff-5c53e0ab0fd1adf5f57c5a054b91eebeR156). I plan to release a fix today.

@IAmAbszol this stuff is documented on the [Replay File Spec](https://github.com/project-slippi/project-slippi/wiki/Replay-File-Spec) wiki page. It's linked from https://github.com/project-slippi/project-slippi, but the link is easy to miss.

**Comment by IAmAbszol** [comment]
@hohav How likely would it be to have a live file streaming option? I see that your game.py uses ubjson which requires the file to be completely finished. I was thinking of splicing the datastream doing it myself but didnt want to intrude on anything you might be in the midst of doing. That page you provided will help a ton as well.

**Comment by hohav** [comment]
@IAmAbszol I'm not working on anything in that area right now— the current file format isn't great for streaming, for the reason you mentioned. I'm sure something could be hacked together, I just don't have much appetite for implementing that.

I remember chatting with Fizzi about possible alternative file formats, but AFAIK nothing came of that.

**Comment by Tanbourine** [comment]
> Thanks for the report. The issue is caused by [a change in Slippi 2.0.0](https://github.com/project-slippi/project-slippi/wiki/Replay-File-Spec/_compare/d3ba3fb88f81e2bb972ff627833473a2fba47f80...1c19dd368d1dd13c8dbabd8e714c6454c138f241#diff-5c53e0ab0fd1adf5f57c5a054b91eebeR156). I plan to release a fix today.
> 
> @IAmAbszol this stuff is documented on the [Replay File Spec](https://github.com/project-slippi/project-slippi/wiki/Replay-File-Spec) wiki page. It's linked from https://github.com/project-slippi/project-slippi, but the link is easy to miss.

Has this issue been addressed? I'm running into this issue still unfortunately (with the same symptoms as above)

Thanks! This is a great tool, I'm really excited to get started

**Comment by hohav** [comment]
@Tanbourine what version were you running? If it was a release, then it wouldn't have had [the fix](https://github.com/hohav/py-slippi/commit/e9c61567e32de333a18038d1faade50c2cc48c8a) as that hadn't been included in a release yet.

I just released 1.3.0, which includes that fix. If it still doesn't work, please create a new ticket.
